internal/providers/oracleoci/oracleoci.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The ec2 provider fetches a remote configuration from the ec2 user-data
+// metadata service URL.
+
+package oracleoci
+
+import (
+	"encoding/base64"
+	"net/url"
+
+	"github.com/coreos/ignition/config/types"
+	"github.com/coreos/ignition/config/validate/report"
+	"github.com/coreos/ignition/internal/providers/util"
+	"github.com/coreos/ignition/internal/resource"
+)
+
+var (
+	userdataUrl = url.URL{
+		Scheme: "http",
+		Host:   "169.254.169.254",
+		Path:   "opc/v1/instance/metadata/user_data",
+	}
+)
+
+func FetchConfig(f resource.Fetcher) (types.Config, report.Report, error) {
+	data, err := f.FetchToBuffer(userdataUrl, resource.FetchOptions{
+		Headers: resource.ConfigHeaders,
+	})
+	if err == resource.ErrNotFound {
```

Comment:
nit

```golang
if err != nil {
  if err == resource.ErrNotFound {
    return types.Config{}, report.Report{}, nil
  }
  return types.Config{}, report.Report{}, err
}
```

arguably allows for better logging management if it ever needs to be added in the future.

Reply:
Can you elaborate? How does this impact logging management?

Reply:
one log statement could be placed after the err != nil to catch the error regardless of ErrNotFound or a real error.

Reply:
In this case `resource.ErrNotFound` isn't an issue. It simply means that the user did not provide an ssh key for this instance. Any log message that would detail an issue here wouldn't apply to the `resource.ErrNotFound` path.